File: app/src/main/java/com/example/octodroid/data/converters/RepositoryConverter.java
package com.example.octodroid.data.converters;

import com.rejasupotaro.octodroid.models.Repository;
import com.rejasupotaro.octodroid.models.Resource;
import com.yatatsu.autobundle.Converter;

public class RepositoryConverter implements Converter<Repository, String> {
    @Override
    public String convert(Repository repository) {
        return repository.toJson();
    }

    @Override
    public Repository original(String serializedRepository) {
        return Resource.fromJson(serializedRepository, Repository.class);
    }
}

File: app/src/main/java/com/example/octodroid/data/converters/RepositoryConverter.kt
package com.example.octodroid.data.converters

import com.rejasupotaro.octodroid.models.Repository
import com.rejasupotaro.octodroid.models.Resource
import com.yatatsu.autobundle.Converter

class RepositoryConverter : Converter<Repository, String> {
    override fun convert(repository: Repository): String {
        return repository.toJson()
    }

    override fun original(serializedRepository: String): Repository {
        return Resource.fromJson(serializedRepository, Repository::class.java)
    }
}
